### PR TITLE
Rework token api

### PR DESCRIPTION
**Pull request by cristaloleg**

### CODE REVIEW

File: example_parse_test.go [hunk 1]
```diff
@@ -6,17 +6,17 @@ import (
 	"github.com/cristalhq/jwt/v3"
 )
 
-func Example_Parse() {
+func ExampleParse() {
```
Comment (FZambia) [inline example_parse_test.go:9]: ParseNoVerify?

File: parse.go [hunk 2]
```diff
@@ -9,21 +9,54 @@ import (
...
+	}
+
+	errVerify := verifier.Verify(tok.PayloadPart(), tok.Signature())
+	if errVerify != nil {
+		return nil, errVerify
+	}
+	return tok, nil
+}
+
+// ParseNoVerifyString decodes a token and verifies it's signature.
```
Comment (FZambia) [inline parse.go:36]: Looks like a comment here is misleading

File: parse.go [hunk 3]
```diff
@@ -9,21 +9,54 @@ import (
...
+func ParseNoVerifyString(token string) (*Token, error) {
+	return ParseNoVerify([]byte(token))
+}
+
+// ParseNoVerify decodes a token and verifies it's signature.
```
Comment (FZambia) [inline parse.go:41]: Also misleading comment

File: audience.go [hunk 4]
```diff
@@ -2,8 +2,9 @@ package jwt
 
 import "encoding/json"
 
-// Audience is a special claim that be a single string or an array of strings
-// see RFC 7519.
+// Audience is a special claim that be a single string or an array of strings.
```
Comment (FZambia) [inline audience.go:5]: Missing word here

File: numeric_date.go [hunk 5]
```diff
@@ -25,7 +25,7 @@ func NewNumericDate(t time.Time) *NumericDate {
 }
 
 // MarshalJSON implements the json.Marshaler interface.
-func (t NumericDate) MarshalJSON() ([]byte, error) {
+func (t *NumericDate) MarshalJSON() ([]byte, error) {
```
Comment (FZambia) [inline numeric_date.go:28]: Wrong comment for NumericDate struct

**Comment by cristaloleg** [comment]
Going to rethink changes in next separate PRs, huge thanks @FZambia

**Pull request closed without merging**